src/Starlight/Actor.hs

```diff
@@ -25,27 +28,31 @@ import Unit.Mass
 import Unit.Time
 
 data Actor = Actor
-  { position :: !(V3 (Kilo Metres Float))
-  , velocity :: !(V3 ((Kilo Metres :/: Seconds) Float))
-  , rotation :: !(Quaternion Float)
-  , mass     :: !(Kilo Grams Float)
+  { position  :: !(V3 (Kilo Metres Float))
+  , velocity  :: !(V3 ((Kilo Metres :/: Seconds) Float))
+  , rotation  :: !(Quaternion Float)
+  , mass      :: !(Kilo Grams Float)
+  , magnitude :: !(Kilo Metres Float) -- approx. equivalent to diameter; should bound the actor’s geometry
   }
   deriving (Generic, Show)
 
-position_ :: Lens' Actor (V3 (Kilo Metres Float))
-position_ = field @"position"
+position_ :: HasCallStack => Lens' Actor (V3 (Kilo Metres Float))
+position_ = field @"position".asserting (none isNaN)
 
-velocity_ :: Lens' Actor (V3 ((Kilo Metres :/: Seconds) Float))
-velocity_ = field @"velocity"
+velocity_ :: HasCallStack => Lens' Actor (V3 ((Kilo Metres :/: Seconds) Float))
+velocity_ = field @"velocity".asserting (none isNaN)
 
-rotation_ :: Lens' Actor (Quaternion Float)
-rotation_ = field @"rotation"
+rotation_ :: HasCallStack => Lens' Actor (Quaternion Float)
+rotation_ = field @"rotation".asserting (none isNaN)
```

Comment:
Checking these for NaN means we throw decently traceable exceptions for the occurrence of NaN on set.